### ISSUE TITLE
body_encoding test seems to randomly "fail"

### ISSUE THREAD

**Issue by robrwo** [sartak/webmachine-perl#29]
Sometimes when running tests, I get the following:

```
t/022-body-encoding.t ............................... All 1 subtests passed 
```

which is treated as a failure:

```
Test Summary Report
-------------------
t/022-body-encoding.t                             (Wstat: 11 Tests: 1 Failed: 0)
  Non-zero wait status: 11
  Parse errors: No plan found in TAP output
```

This failure goes away when re-running the tests, sometimes.

**Comment by autarch** [comment]
I can't reproduce this. I'm guessing it's some difference in the prereq chain somewhere.

**Comment by eserte** [comment]
These are segmentation faults. CPAN Testers has also some of these:
http://analysis.cpantesters.org/reports_by_field?SUBMIT_xxx=Submit&distv=Web-Machine-0.16&field=fail%3At%2F022-body-encoding.t&field=fail%3At%2F022-body-encoding.t&field=meta%3Aperl
I see a higher probability for a perl 5.20.x to get such a fail.

Maybe related: 302-content-negotiation-charset.t also has random segfaults:
http://analysis.cpantesters.org/reports_by_field?SUBMIT_xxx=Submit&distv=Web-Machine-0.16&field=fail%3At%2F302-content-negotiation-charset.t&field=fail%3At%2F302-content-negotiation-charset.t&field=meta%3Aperl

**Comment by autarch** [comment]
Yeah, I've seen this come in over the months but I'm not sure what's causing it. There's not many XS modules in the dep tree for this distro, AFAICT.

**Comment by eserte** [comment]
`302-content-negotiation-charset.t` is caused by a problem in `Encode`, see https://rt.perl.org/Public/Bug/Display.html?id=122873 and which should be fixed in newer `Encode` versions. Maybe `022-body-encoding.t` has the same problem? @robrwo: do you still see the errors, and if yes, what `Encode` version do you have installed?
